packages/mds-ingest-service/repository/index.ts

```diff
@@ -472,20 +472,30 @@ class IngestReadWriteRepository extends ReadWriteRepository {
 
   public getLatestTelemetryForDevices = async (device_ids: UUID[]): Promise<TelemetryDomainModel[]> => {
     try {
-      const connection = await this.connect('ro')
-      const entities = await connection
-        .createQueryBuilder(TelemetryEntity, 'telemetry')
-        .innerJoin(
-          subquery =>
-            subquery
-              .select('id, RANK() OVER (PARTITION BY device_id ORDER BY timestamp DESC) AS rownum')
-              .where('device_id = ANY(:device_ids)', { device_ids })
-              .from(TelemetryEntity, 't'),
-          'last_device_telemetry',
-          'last_device_telemetry.id = telemetry.id AND last_device_telemetry.rownum = 1'
-        )
-        .getMany()
-      return entities.map(TelemetryEntityToDomain.mapper())
+      /**
+       * https://github.com/typeorm/typeorm/blob/master/docs/transactions.md#using-queryrunner-to-create-and-control-state-of-single-database-connection
+       */
+      const runner = await (await this.connect('ro')).createQueryRunner()
+
+      await runner.startTransaction()
+      try {
+        /* Docs: set_config sets the parameter setting_name to new_value. If is_local is true, the new value will only apply to the current transaction
+           Sets enable_bitmapscan off for this transaction/query
+        */
+        await runner.query(`select set_config('enable_bitmapscan','off', true)`)
```

Comment:
I do wonder if this works because of something else we’re not doing or doing wrong such that the planner doesn’t do the correct thing. Are the db statistics outdated/inaccurate?, etc. We basically don’t do any sort of regular db maintenance.

Reply:
Maybe we want a follow-up (tech debt?) ticket to look at stats.

I would be very surprised if this set_config didn't work on an ro connection.

Reply:
I agree @PeterSP ... Neil and I tested this on develop yesterday and it's configured with a read replica so I think set_config does indeed work.

Reply:
I did a test, on a RO replica in prod, and you _can_ in fact use `select set_config()`. You just can't do create/insert/delete.